### PR TITLE
Switch from restore to publish

### PR DESCRIPTION
**Pull request by NCarlsonMSFT**
Changes to building ASP.NET RC2 projects on the dev machine and publishing them into the image rather than building within the image.

### CODE REVIEW

File: generators/app/index.js [hunk 1]
```diff
@@ -178,29 +179,51 @@ function handleAspNet(yo) {
...
             yo.log.error(err);
+            done();
             return;
         }
-        configureUrlsNoteForUser = noteForUser;
-        done();
+        updateProjectJsonNoteForUser = projectJsonNote;
+
+        if (aspNet.getDotnetVersion() == 'RC2') {
+            aspNet.updateProgramCS(function (err, programCSNode) {
```
Comment (saurabsa) [inline generators/app/index.js]: Why is updateProgramCS under updateProjectJson? Shouldn't they be processed separately?

File: generators/app/templates/_dockerTaskGeneric.sh [hunk 2]
```diff
@@ -26,10 +26,18 @@ buildImage () {
       taggedImageName="$imageName"
       if [[ $ENVIRONMENT != "release" ]]; then
         taggedImageName="$imageName:$ENVIRONMENT"
-      fi
+      fi<% if (projectType === 'aspnet' && dotnetVersion === 'RC2') { %>
+
+      echo "Building the project ($ENVIRONMENT)."
+      pubFolder="bin/$ENVIRONMENT/netcoreapp1.0/publish"
+      dotnet build -c $ENVIRONMENT
+      dotnet publish -c $ENVIRONMENT -o $pubFolder
```
Comment (saurabsa) [inline generators/app/templates/_dockerTaskGeneric.sh]: Don't we need to provide framework in the publish command?

File: generators/app/aspnetHelper.js [hunk 3]
```diff
@@ -91,46 +117,108 @@ AspNetHelper.prototype.configureUrls = function (cb) {
...
+            if (data.buildOptions !== "portable") {
+                data.buildOptions.debugType = "portable";
+                changed = true;
+            }
+            if (data.publishOptions === undefined) {
+                data.publishOptions = { };
+                changed = true;
+            }
+            if (data.publishOptions.include === undefined) {
+                data.publishOptions.include = [ ];
```
Comment (saurabsa) [inline generators/app/aspnetHelper.js]: The publishOptions should include 
     [ "docker-compose.yml",
      "docker-compose.debug.yml",
      "dockerfile.debug",
      "dockerfile" ]

File: generators/app/aspnetHelper.js [hunk 4]
```diff
@@ -91,46 +117,108 @@ AspNetHelper.prototype.configureUrls = function (cb) {
...
+        // Remove BOM.
+        if (data.charCodeAt(0) === 0xFEFF) {
+            data = data.replace(/^\uFEFF/, '');
+        }
+
+        data = JSON.parse(data);
+
+        if (dotnetVersion === 'RC2' ) {
+            var changed = false;
+            if (data.buildOptions === undefined) {
```
Comment (PeterJausovec) [inline generators/app/aspnetHelper.js]: nit: you could replace `data.buildOptions === undefined` with `!data.buildOptions`

File: generators/app/templates/_dockerTaskGeneric.sh (hunk 2 above)
Comment (NCarlsonMSFT) [inline generators/app/templates/_dockerTaskGeneric.sh]: Apparently not as it worked ;) But I agree that would be better form.

---
In reply to: [67467303](https://github.com/Microsoft/generator-docker/pull/69#discussion_r67467303) [](ancestors = 67467303)

File: generators/app/aspnetHelper.js (hunk 3 above)
Comment (NCarlsonMSFT) [inline generators/app/aspnetHelper.js]: There is no need to publish the compose files as they only reference images so their location does not matter.

Also i initialize it to empty so the logic below can populate it, maintaining the list of files to publish in two places has bit us before. 

---
In reply to: [67468521](https://github.com/Microsoft/generator-docker/pull/69#discussion_r67468521) [](ancestors = 67468521)

File: generators/app/aspnetHelper.js (hunk 4 above)
Comment (NCarlsonMSFT) [inline generators/app/aspnetHelper.js]: #Pending

---
In reply to: [67532250](https://github.com/Microsoft/generator-docker/pull/69#discussion_r67532250) [](ancestors = 67532250)

File: generators/app/index.js (hunk 1 above)
Comment (NCarlsonMSFT) [inline generators/app/index.js]: They are being process separately. updateProgramCS is being called as part of the continuation of calling updateProjectJson. You can see a similar pattern of continuing in the call-back in both updateProjectJson and updateProgramCS's implementations

---
In reply to: [67465190](https://github.com/Microsoft/generator-docker/pull/69#discussion_r67465190) [](ancestors = 67465190)

**Comment by PeterJausovec** [comment]
Looks good to me!

**Comment by saurabsa** [comment]
:shipit:

**Pull request merged**